// @kingstinct/react-native-healthkit/lib/module/types/Workouts.js
export var WorkoutActivityType;
(function (WorkoutActivityType) {
    WorkoutActivityType[WorkoutActivityType["americanFootball"] = 1] = "americanFootball";
    WorkoutActivityType[WorkoutActivityType["archery"] = 2] = "archery";
    WorkoutActivityType[WorkoutActivityType["australianFootball"] = 3] = "australianFootball";
    WorkoutActivityType[WorkoutActivityType["badminton"] = 4] = "badminton";
    WorkoutActivityType[WorkoutActivityType["baseball"] = 5] = "baseball";
    WorkoutActivityType[WorkoutActivityType["basketball"] = 6] = "basketball";
    WorkoutActivityType[WorkoutActivityType["bowling"] = 7] = "bowling";
    WorkoutActivityType[WorkoutActivityType["boxing"] = 8] = "boxing";
    WorkoutActivityType[WorkoutActivityType["climbing"] = 9] = "climbing";
    WorkoutActivityType[WorkoutActivityType["cricket"] = 10] = "cricket";
    WorkoutActivityType[WorkoutActivityType["crossTraining"] = 11] = "crossTraining";
    WorkoutActivityType[WorkoutActivityType["curling"] = 12] = "curling";
    WorkoutActivityType[WorkoutActivityType["cycling"] = 13] = "cycling";
    WorkoutActivityType[WorkoutActivityType["dance"] = 14] = "dance";
    WorkoutActivityType[WorkoutActivityType["danceInspiredTraining"] = 15] = "danceInspiredTraining";
    WorkoutActivityType[WorkoutActivityType["elliptical"] = 16] = "elliptical";
    WorkoutActivityType[WorkoutActivityType["equestrianSports"] = 17] = "equestrianSports";
    WorkoutActivityType[WorkoutActivityType["fencing"] = 18] = "fencing";
    WorkoutActivityType[WorkoutActivityType["fishing"] = 19] = "fishing";
    WorkoutActivityType[WorkoutActivityType["functionalStrengthTraining"] = 20] = "functionalStrengthTraining";
    WorkoutActivityType[WorkoutActivityType["golf"] = 21] = "golf";
    WorkoutActivityType[WorkoutActivityType["gymnastics"] = 22] = "gymnastics";
    WorkoutActivityType[WorkoutActivityType["handball"] = 23] = "handball";
    WorkoutActivityType[WorkoutActivityType["hiking"] = 24] = "hiking";
    WorkoutActivityType[WorkoutActivityType["hockey"] = 25] = "hockey";
    WorkoutActivityType[WorkoutActivityType["hunting"] = 26] = "hunting";
    WorkoutActivityType[WorkoutActivityType["lacrosse"] = 27] = "lacrosse";
    WorkoutActivityType[WorkoutActivityType["martialArts"] = 28] = "martialArts";
    WorkoutActivityType[WorkoutActivityType["mindAndBody"] = 29] = "mindAndBody";
    WorkoutActivityType[WorkoutActivityType["mixedMetabolicCardioTraining"] = 30] = "mixedMetabolicCardioTraining";
    WorkoutActivityType[WorkoutActivityType["paddleSports"] = 31] = "paddleSports";
    WorkoutActivityType[WorkoutActivityType["play"] = 32] = "play";
    WorkoutActivityType[WorkoutActivityType["preparationAndRecovery"] = 33] = "preparationAndRecovery";
    WorkoutActivityType[WorkoutActivityType["racquetball"] = 34] = "racquetball";
    WorkoutActivityType[WorkoutActivityType["rowing"] = 35] = "rowing";
    WorkoutActivityType[WorkoutActivityType["rugby"] = 36] = "rugby";
    WorkoutActivityType[WorkoutActivityType["running"] = 37] = "running";
    WorkoutActivityType[WorkoutActivityType["sailing"] = 38] = "sailing";
    WorkoutActivityType[WorkoutActivityType["skatingSports"] = 39] = "skatingSports";
    WorkoutActivityType[WorkoutActivityType["snowSports"] = 40] = "snowSports";
    WorkoutActivityType[WorkoutActivityType["soccer"] = 41] = "soccer";
    WorkoutActivityType[WorkoutActivityType["softball"] = 42] = "softball";
    WorkoutActivityType[WorkoutActivityType["squash"] = 43] = "squash";
    WorkoutActivityType[WorkoutActivityType["stairClimbing"] = 44] = "stairClimbing";
    WorkoutActivityType[WorkoutActivityType["surfingSports"] = 45] = "surfingSports";
    WorkoutActivityType[WorkoutActivityType["swimming"] = 46] = "swimming";
    WorkoutActivityType[WorkoutActivityType["tableTennis"] = 47] = "tableTennis";
    WorkoutActivityType[WorkoutActivityType["tennis"] = 48] = "tennis";
    WorkoutActivityType[WorkoutActivityType["trackAndField"] = 49] = "trackAndField";
    WorkoutActivityType[WorkoutActivityType["traditionalStrengthTraining"] = 50] = "traditionalStrengthTraining";
    WorkoutActivityType[WorkoutActivityType["volleyball"] = 51] = "volleyball";
    WorkoutActivityType[WorkoutActivityType["walking"] = 52] = "walking";
    WorkoutActivityType[WorkoutActivityType["waterFitness"] = 53] = "waterFitness";
    WorkoutActivityType[WorkoutActivityType["waterPolo"] = 54] = "waterPolo";
    WorkoutActivityType[WorkoutActivityType["waterSports"] = 55] = "waterSports";
    WorkoutActivityType[WorkoutActivityType["wrestling"] = 56] = "wrestling";
    WorkoutActivityType[WorkoutActivityType["yoga"] = 57] = "yoga";
    WorkoutActivityType[WorkoutActivityType["barre"] = 58] = "barre";
    WorkoutActivityType[WorkoutActivityType["coreTraining"] = 59] = "coreTraining";
    WorkoutActivityType[WorkoutActivityType["crossCountrySkiing"] = 60] = "crossCountrySkiing";
    WorkoutActivityType[WorkoutActivityType["downhillSkiing"] = 61] = "downhillSkiing";
    WorkoutActivityType[WorkoutActivityType["flexibility"] = 62] = "flexibility";
    WorkoutActivityType[WorkoutActivityType["highIntensityIntervalTraining"] = 63] = "highIntensityIntervalTraining";
    WorkoutActivityType[WorkoutActivityType["jumpRope"] = 64] = "jumpRope";
    WorkoutActivityType[WorkoutActivityType["kickboxing"] = 65] = "kickboxing";
    WorkoutActivityType[WorkoutActivityType["pilates"] = 66] = "pilates";
    WorkoutActivityType[WorkoutActivityType["snowboarding"] = 67] = "snowboarding";
    WorkoutActivityType[WorkoutActivityType["stairs"] = 68] = "stairs";
    WorkoutActivityType[WorkoutActivityType["stepTraining"] = 69] = "stepTraining";
    WorkoutActivityType[WorkoutActivityType["wheelchairWalkPace"] = 70] = "wheelchairWalkPace";
    WorkoutActivityType[WorkoutActivityType["wheelchairRunPace"] = 71] = "wheelchairRunPace";
    WorkoutActivityType[WorkoutActivityType["taiChi"] = 72] = "taiChi";
    WorkoutActivityType[WorkoutActivityType["mixedCardio"] = 73] = "mixedCardio";
    WorkoutActivityType[WorkoutActivityType["handCycling"] = 74] = "handCycling";
    WorkoutActivityType[WorkoutActivityType["discSports"] = 75] = "discSports";
    WorkoutActivityType[WorkoutActivityType["fitnessGaming"] = 76] = "fitnessGaming";
    WorkoutActivityType[WorkoutActivityType["cardioDance"] = 77] = "cardioDance";
    WorkoutActivityType[WorkoutActivityType["socialDance"] = 78] = "socialDance";
    WorkoutActivityType[WorkoutActivityType["pickleball"] = 79] = "pickleball";
    WorkoutActivityType[WorkoutActivityType["cooldown"] = 80] = "cooldown";
    WorkoutActivityType[WorkoutActivityType["swimBikeRun"] = 82] = "swimBikeRun";
    WorkoutActivityType[WorkoutActivityType["transition"] = 83] = "transition";
    WorkoutActivityType[WorkoutActivityType["underwaterDiving"] = 84] = "underwaterDiving";
    WorkoutActivityType[WorkoutActivityType["other"] = 3000] = "other";
})(WorkoutActivityType || (WorkoutActivityType = {}));
// documented at https://developer.apple.com/documentation/healthkit/hkweathercondition
export var WeatherCondition;
(function (WeatherCondition) {
    WeatherCondition[WeatherCondition["none"] = 0] = "none";
    WeatherCondition[WeatherCondition["clear"] = 1] = "clear";
    WeatherCondition[WeatherCondition["fair"] = 2] = "fair";
    WeatherCondition[WeatherCondition["partlyCloudy"] = 3] = "partlyCloudy";
    WeatherCondition[WeatherCondition["mostlyCloudy"] = 4] = "mostlyCloudy";
    WeatherCondition[WeatherCondition["cloudy"] = 5] = "cloudy";
    WeatherCondition[WeatherCondition["foggy"] = 6] = "foggy";
    WeatherCondition[WeatherCondition["haze"] = 7] = "haze";
    WeatherCondition[WeatherCondition["windy"] = 8] = "windy";
    WeatherCondition[WeatherCondition["blustery"] = 9] = "blustery";
    WeatherCondition[WeatherCondition["smoky"] = 10] = "smoky";
    WeatherCondition[WeatherCondition["dust"] = 11] = "dust";
    WeatherCondition[WeatherCondition["snow"] = 12] = "snow";
    WeatherCondition[WeatherCondition["hail"] = 13] = "hail";
    WeatherCondition[WeatherCondition["sleet"] = 14] = "sleet";
    WeatherCondition[WeatherCondition["freezingDrizzle"] = 15] = "freezingDrizzle";
    WeatherCondition[WeatherCondition["freezingRain"] = 16] = "freezingRain";
    WeatherCondition[WeatherCondition["mixedRainAndHail"] = 17] = "mixedRainAndHail";
    WeatherCondition[WeatherCondition["mixedRainAndSnow"] = 18] = "mixedRainAndSnow";
    WeatherCondition[WeatherCondition["mixedRainAndSleet"] = 19] = "mixedRainAndSleet";
    WeatherCondition[WeatherCondition["mixedSnowAndSleet"] = 20] = "mixedSnowAndSleet";
    WeatherCondition[WeatherCondition["drizzle"] = 21] = "drizzle";
    WeatherCondition[WeatherCondition["scatteredShowers"] = 22] = "scatteredShowers";
    WeatherCondition[WeatherCondition["showers"] = 23] = "showers";
    WeatherCondition[WeatherCondition["thunderstorms"] = 24] = "thunderstorms";
    WeatherCondition[WeatherCondition["tropicalStorm"] = 25] = "tropicalStorm";
    WeatherCondition[WeatherCondition["hurricane"] = 26] = "hurricane";
    WeatherCondition[WeatherCondition["tornado"] = 27] = "tornado";
})(WeatherCondition || (WeatherCondition = {}));
export var WorkoutEventType;
(function (WorkoutEventType) {
    WorkoutEventType[WorkoutEventType["pause"] = 1] = "pause";
    WorkoutEventType[WorkoutEventType["resume"] = 2] = "resume";
    WorkoutEventType[WorkoutEventType["lap"] = 3] = "lap";
    WorkoutEventType[WorkoutEventType["marker"] = 4] = "marker";
    WorkoutEventType[WorkoutEventType["motionPaused"] = 5] = "motionPaused";
    WorkoutEventType[WorkoutEventType["motionResumed"] = 6] = "motionResumed";
    WorkoutEventType[WorkoutEventType["segment"] = 7] = "segment";
    WorkoutEventType[WorkoutEventType["pauseOrResumeRequest"] = 8] = "pauseOrResumeRequest";
})(WorkoutEventType || (WorkoutEventType = {}));
var ComparisonPredicateOperator;
(function (ComparisonPredicateOperator) {
    ComparisonPredicateOperator[ComparisonPredicateOperator["lessThan"] = 0] = "lessThan";
    ComparisonPredicateOperator[ComparisonPredicateOperator["lessThanOrEqualTo"] = 1] = "lessThanOrEqualTo";
    ComparisonPredicateOperator[ComparisonPredicateOperator["greaterThan"] = 2] = "greaterThan";
    ComparisonPredicateOperator[ComparisonPredicateOperator["greaterThanOrEqualTo"] = 3] = "greaterThanOrEqualTo";
    ComparisonPredicateOperator[ComparisonPredicateOperator["equalTo"] = 4] = "equalTo";
    ComparisonPredicateOperator[ComparisonPredicateOperator["notEqualTo"] = 5] = "notEqualTo";
    ComparisonPredicateOperator[ComparisonPredicateOperator["matches"] = 6] = "matches";
    ComparisonPredicateOperator[ComparisonPredicateOperator["like"] = 7] = "like";
    ComparisonPredicateOperator[ComparisonPredicateOperator["beginsWith"] = 8] = "beginsWith";
    ComparisonPredicateOperator[ComparisonPredicateOperator["endsWith"] = 9] = "endsWith";
    ComparisonPredicateOperator[ComparisonPredicateOperator["in"] = 10] = "in";
    ComparisonPredicateOperator[ComparisonPredicateOperator["customSelector"] = 11] = "customSelector";
    ComparisonPredicateOperator[ComparisonPredicateOperator["contains"] = 99] = "contains";
    ComparisonPredicateOperator[ComparisonPredicateOperator["between"] = 100] = "between";
})(ComparisonPredicateOperator || (ComparisonPredicateOperator = {}));
